### PR TITLE
i18n(de): update translation guides/i18n

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description

Translation of `guides/i18n.mdx` into German language.

### CODE REVIEW

**Comment by trueberryless** [comment]
Links fixed in #2476 and #2459

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 1]
```diff
@@ -55,21 +57,25 @@ Starlight bietet eingebaute Unterstützung für mehrsprachige Seiten, einschlie
...
 
    </FileTree>
 
 3. Du kannst nun Inhaltsdateien in deinen Sprachverzeichnissen hinzufügen. Verwende den gleichen Dateinamen, um Seiten in verschiedenen Sprachen zuzuordnen und nutze Starlights volle i18n-Funktionen, einschließlich Fallback-Inhalte, Übersetzungshinweise und mehr.
 
    Erstelle zum Beispiel `ar/index.md` und `en/index.md`, um die Homepage für Arabisch bzw. Englisch darzustellen.
 
+</Steps>
+
+Für fortgeschrittene i18n-Szenarien unterstützt Starlight auch die Konfiguration der Internationalisierung mit der Option [Astro's `i18n` config](https://docs.astro.build/de/guides/internationalization/#i18n-routing-konfigurieren).
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:70]: Just to double check, is "Astro's `i18n` config" expected or something that could have been missed? Asking as it's identical to the English version.

Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:70]:
```suggestion
Für fortgeschrittene i18n-Szenarien unterstützt Starlight auch die Konfiguration der Internationalisierung mit der Option [Astro's `i18n` config](https://docs.astro.build/de/guides/internationalization/#configure-i18n-routing).
```

Invalid anchor

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 2]
```diff
@@ -223,3 +248,159 @@ Du kannst Übersetzungen für zusätzliche Sprachen, die du unterstützt, über
...
+		schema: i18nSchema({
++			extend: z.object({
++				'custom.label': z.string().optional(),
++			}),
+		}),
+	}),
+};
+```
+
+Mehr über Inhaltssammlungsschemata erfährst du in [„Defining a collection schema“](https://docs.astro.build/de/guides/content-collections/#definieren-eines-sammelschemas) in den Astro-Dokumenten.
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:277]: I assume the "Defining a collection schema" link name was forgotten.

Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:277]:
```suggestion
Mehr über Inhaltssammlungsschemata erfährst du in [„Defining a collection schema“](https://docs.astro.build/de/guides/content-collections/#defining-a-collection-schema) in den Astro-Dokumenten.
```

Invalid anchor

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 3]
```diff
@@ -223,3 +248,159 @@ Du kannst Übersetzungen für zusätzliche Sprachen, die du unterstützt, über
...
+
+In Astro-Komponenten ist diese API als Teil des [globalen `Astro`-Objekts](https://docs.astro.build/de/reference/api-reference/#astrolocals) als `Astro.locals.t` verfügbar:
+
+```astro title="example.astro"
+<p dir={Astro.locals.t.dir()}>
+	{Astro.locals.t('404.text')}
+</p>
+```
+
+Du kannst die API auch bei [Endpunkten](https://docs.astro.build/de/guides/endpoints/) verwenden, wo das Objekt `locals` als Teil des [Endpunkt-Kontextes] (https://docs.astro.build/de/reference/api-reference/#contextlocals) verfügbar ist:
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:293]:
```suggestion
Du kannst die API auch bei [Endpunkten](https://docs.astro.build/de/guides/endpoints/) verwenden, wo das Objekt `locals` als Teil des [Endpunkt-Kontextes](https://docs.astro.build/de/reference/api-reference/#contextlocals) verfügbar ist:
```

Invalid link markup (extra space).

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 4]
```diff
@@ -223,3 +248,159 @@ Du kannst Übersetzungen für zusätzliche Sprachen, die du unterstützt, über
...
+```
+
+Der erste UI-String kann gerendert werden, indem man `'link.astro'` an die Funktion `t()` übergibt:
+
+```astro {3}
+<!-- src/components/Example.astro -->
+<a href="https://docs.astro.build/">
+	{Astro.locals.t('link.astro')}
+</a>
+<!-- Renders: <a href="...">Astro Dokumentation</a> -->
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:322]: Is "Renders" expected here or missed from the original translation?

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 5]
```diff
@@ -223,3 +248,159 @@ Du kannst Übersetzungen für zusätzliche Sprachen, die du unterstützt, über
...
+
+Der zweite UI-String verwendet die [Interpolationssyntax](https://www.i18next.com/translation-function/interpolation) von i18next für den Platzhalter `{{feature}}`.
+Der Wert für `feature` muss in einem Optionsobjekt gesetzt werden, das als zweites Argument an `t()` übergeben wird:
+
+```astro {3}
+<!-- src/components/Example.astro -->
+<a href="https://docs.astro.build/en/guides/astro-db/">
+	{Astro.locals.t('link.astro.custom', { feature: 'Astro DB' })}
+</a>
+<!-- Renders: <a href="...">Astro-Dokumentation für Astro DB</a> -->
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:333]: Same here for "Renders"

File: docs/src/content/docs/de/guides/i18n.mdx [hunk 6]
```diff
@@ -223,3 +248,159 @@ Du kannst Übersetzungen für zusätzliche Sprachen, die du unterstützt, über
...
+	}),
+};
+```
+
+Mehr über Inhaltssammlungsschemata erfährst du in [„Defining a collection schema“](https://docs.astro.build/de/guides/content-collections/#definieren-eines-sammelschemas) in den Astro-Dokumenten.
+
+
+## UI-Übersetzungen verwenden
+
+Du kannst auf Starlights [eingebaute UI-Strings](/de/guides/i18n/#starlights-ui-%C3%BCbersetzen) sowie auf [benutzerdefinierte](/de/guides/i18n/#%C3%BCbersetzungsschema-erweitern) und [plugin-provided](/de/reference/plugins/#injecttranslations) UI-Strings über eine einheitliche API zugreifen, die von [i18next](https://www.i18next.com/) unterstützt wird.
```
Comment (HiDeoo) [inline docs/src/content/docs/de/guides/i18n.mdx:282]:
```suggestion
Du kannst auf Starlights [eingebaute UI-Strings](/de/guides/i18n/#starlights-ui-übersetzen) sowie auf [benutzerdefinierte](/de/guides/i18n/#übersetzungsschema-erweitern) und [plugin-provided](/de/reference/plugins/#injecttranslations) UI-Strings über eine einheitliche API zugreifen, die von [i18next](https://www.i18next.com/) unterstützt wird.
```

This should fix the same-page link errors.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the translation :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://contribute.docs.astro.build/reviewers/reviewing-translations/). I left a few suggestions and questions but other than that, the translation looks good to me.

**Comment by trueberryless** [comment]
Thanks for your review, sry that I missed that many translations...

**Review by HiDeoo — APPROVED** [review]
Thanks for updating the PR, should be good to go :rocket:

**Pull request merged**